graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/ElasticsearchClient.java

```diff
@@ -146,4 +148,14 @@ private boolean isInvalidWriteTargetException(ElasticsearchException elasticsear
     private boolean isIndexNotFoundException(ElasticsearchException e) {
         return e.getMessage().contains("index_not_found_exception");
     }
+
+    public static RequestOptions withTimeout(RequestOptions requestOptions, Duration timeout) {
+        final RequestConfig requestConfigWithTimeout = RequestConfig.copy(requestOptions.getRequestConfig())
+                .setSocketTimeout(Math.toIntExact(timeout.toMilliseconds()))
```

Comment:
How about using `Ints.saturatedCast(long)` here to avoid an exception if the timeout overflows an int?

Reply:
I am not sure. This would mean that we would be silently doing something else than requested. Alternatively we could validate the config setting to not be higher than 24 days (~`Integer.MAX_VALUE / (24*60*60*1000)`)?

Reply:
Sounds good. :+1: I guess we can also choose a lower maximum. I doubt that someone wants to have such a high timeout. :smile:

Reply:
That is true, but I am too uncreative right now to come up with something different, so I just picked 24 days :)